### PR TITLE
OCPBUGS-27948: Tuned rendering and MCP detection improvements

### PR DESCRIPTION
**Pull request by MarSik**
This is a backport of three related PRs: #878, #833 and #924 that improve the rendering mode during the installation bootstrap phase.

The most notable change is the added assumption that basic master and worker MachineConfigPools with default labels will always exist even though they are not yet available in the input manifest folder.

### CODE REVIEW

**Comment by MarSik** [comment]
@jlojosnegros @vitus133 @eggfoobar Took a me a while to untangle the patch ordering, but this should be it.

File: hack/render-sync.sh [hunk 1]
```diff
@@ -1,16 +1,35 @@
...
---asset-output-dir "${ARTIFACT_DIR}"
-
-cp "${ARTIFACT_DIR}"/*  "${WORKDIR}"/test/e2e/performanceprofile/testdata/render-expected-output/default
-for f in  "${WORKDIR}"/test/e2e/performanceprofile/testdata/render-expected-output/default/*
-do
-  sed -i "s/uid:.*/uid: \"\"/" "${f}"
-done
-rm -r "${ARTIFACT_DIR}"
+
+function join_by { local IFS="$1"; shift; echo "$*"; }
```
[the remark this answers is not in the document]
Comment (MarSik) [reply hack/render-sync.sh:4]: I did not invent this one actually. But it uses bash features cleverly indeed.

**Comment by MarSik** [comment]
> I guess there's no simple way to test before to commit.

Actually there is a way. I executed the code and it generated the exact same manifests we already had in the repo. git reported no changes. I also executed the e2e local tests that validate the rendering and all passed. That probably means the output is still valid and matches the rendered yamls.

File: pkg/tuned/cmd/render/render.go [hunk 2]
```diff
@@ -126,17 +129,40 @@ func render(inputDir []string, outputDir string) error {
...
+		return fmt.Errorf("Unable to find MachineConfigPool:%q in input folders", mcpName)
+	}
+
+	filteredPerformanceProfiles, err := filterPerformanceProfilesByMachineConfigPool(perfProfiles, mcp)
+	if err != nil {
+		klog.Errorf("Unable to find a PerformanceProfile that matches the MachineConfigPool %s. error : %v", mcpName, err)
+		return fmt.Errorf("Unable to get PerformanceProfile to apply using MachineConfigPool %s. error : %w", mcpName, err)
+	}
+
+	if len(filteredPerformanceProfiles) == 0 {
```
Comment (jlojosnegros) [inline pkg/tuned/cmd/render/render.go:147]: We had just changed this behaviour in master with #924 so the absence of a performance profile is not considered an error.

**Pull request merged**